Mathlib/RingTheory/MvPowerSeries/Topology.lean

```diff
@@ -0,0 +1,311 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos Fernández
+-/
+
+import Mathlib.RingTheory.Nilpotent.Defs
+import Mathlib.RingTheory.MvPowerSeries.Basic
+import Mathlib.Topology.Algebra.InfiniteSum.Constructions
+import Mathlib.Topology.Algebra.Ring.Basic
+import Mathlib.Topology.Algebra.UniformGroup
+import Mathlib.Topology.UniformSpace.Pi
+
+/-! # Topology on power series
+
+Let `α` be with Semiring α` and `TopologicalSpace α`
+In this file we the topology on `MvPowerSeries σ α`
+that corresponds to the simple convergence on its coefficients.
+It is the coarsest topology for which all coefficients maps are continuous.
+
+When `α` has `UniformSpace α`, we define the corresponding uniform structure.
+
+When the type of coefficients has the discrete topology,
+it corresponds to the topology defined by [bourbaki1981], chapter 4, §4, n°2
+
+It is *not* the adic topology in general.
+
+- `tendsto_pow_zero_of_constantCoeff_nilpotent`, `tendsto_pow_zero_of_constantCoeff_zero`:
+if the constant coefficient of `f` is nilpotent, or vanishes,
+then the powers of `f` converge to zero.
+
+- `tendsto_pow_of_constantCoeff_nilpotent_iff` : the powers of `f` converge to zero iff
+the constant coefficient of `f` is nilpotent
+
+- `hasSum_of_monomials_self` : viewed as an infinite sum, a power series coverges to itself
+
+TODO: add the similar result for the series of homogeneous components
+
+## Instances
+
+- If `α` is a topological (semi)ring, then so is `MvPowerSeries σ α`
+
+- If the topology of `α` is T2, then so is that of `MvPowerSeries σ α``
+
+- If `α` is a `uniformAddGroup`, then so is `MvPowerSeries σ α``
+
+- If `α` is complete, then so is `MvPowerSeries σ α`
+
+-/
+
+
+theorem Finset.prod_one_add {ι α : Type _} [DecidableEq ι] [CommRing α] {f : ι → α} (s : Finset ι) :
+    (s.prod fun i => 1 + f i) = s.powerset.sum fun t => t.prod f := by
+  simp_rw [add_comm, Finset.prod_add]
+  congr
+  ext t
+  convert mul_one (Finset.prod t fun a => f a)
+  exact Finset.prod_eq_one (fun i _ => rfl)
+
+theorem MvPowerSeries.coeff_eq_apply {σ α : Type _} [Semiring α] (f : MvPowerSeries σ α)
+    (d : σ →₀ ℕ) : MvPowerSeries.coeff α d f = f d :=
+  rfl
+
+namespace MvPowerSeries
+
+open Function
+
+variable (σ : Type _) (α : Type _)
+
+section Topological
+
+variable [TopologicalSpace α]
+
+namespace WithPiTopology
```

Comment:
Do you need the `section`, or is the `namespace` enough?
```suggestion
namespace WithPiTopology

variable [TopologicalSpace α]
```

Reply:
namespace should be enough (they keep variables within the namespace… end, but that wasn't so clear a few months ago)